test/test_operations.py

```diff
@@ -198,8 +198,11 @@ def runAtenTest(self, tensors, fn, device=None, rel_err=1e-2, abs_err=1e-5):
     results = xu.as_list(fn(*tensors))
     xla_results = xu.as_list(fn(*xla_tensors))
     for at, xt in zip(results, xla_results):
-      self.assertEqualRel(
-          self.makeComparable(xt), at, rel_err=rel_err, abs_err=abs_err)
+      xt = self.makeComparable(xt)
```

Comment:
It does, the problem is `abs()`, which isn't allowed by PyTorch for boolean tensors: `diff_tensor = (out - expected).abs().float()`. I thought about switching the `float()` and the `abs()` and it might work, but looking for relative errors on boolean is a bit weird anyway.

Reply:
This is a generic `runAtenTest()` API, which already has `rel_err` and `abs_err`.
So given that the change in switching `abs()` and `float()` is minimal, I'd do that instead of other generic (like `runAtenTest()` - which can test anything) APIs having to replicate the `if` below.
I would not want to have a `runIntegralAtenTest()` and `runFloatingPointAtentest()` either at this point.
Python is a bit different from C++, so I think the `assertEqualRel()` handling makes sense.

Reply:
There are additional issues when dumping differences, so I chose to convert to byte tensors in `makeComparable` (and call it for the ATen reference as well).

Reply:
While I do not mind the change, I would mind in a code review, "additional issues" being expanded into some more meaningful comment, so the reviewers can understand, and eventually replicate, the mentioned issues.

Reply:
I meant the same class of issues we see in the equality check, we repeat the computations in `assertEqualRel` to identify which element differences need to be printed. Tracking down what `_dump_differences` would do in case of a difference for bool tensors, we go into the `check_values` closure, which immediately does `abs`: https://github.com/pytorch/xla/blob/master/test/test_operations.py#L115.